src/Controller/Form.php

```diff
@@ -64,6 +62,11 @@ public function loadTemplateOnFrontend() {
 	 * @since 2.7.0
 	 */
 	public function loadReceiptView() {
+		// Do not handle legacy donation form.
+		if ( FormUtils::isLegacyForm() ) {
+			return;
```

Comment:
Should this return false?

Reply:
Looks like this method is intended to `exit()` and therefore return void, so returning void is fine.

That said, @ravinderk, would it make more sense for `loadReceiptView` not to be called in a legacy context, rather than making this method legacy-aware? That's a genuine question, not an assertion; this may be the best place.

Reply:
@bradp if you check function calling sequence then you will find that to detect form id we are using `get_query_var` function will give a fatal error if used before `wp` hook trigger. I think we need a separate issue to discuss embed form loading flow.

Reply:
@JasonTheAdams I think also need a class to handle router and controller after that we do not need to add a hacky code like this.